PSReadLine/Render.cs

```diff
@@ -310,38 +311,61 @@ void MaybeEmphasize(int i, string currColor)
                     activeColor = string.Empty;
 
                     UpdateColorsIfNecessary(Options._continuationPromptColor);
-                    foreach (char c in Options.ContinuationPrompt)
-                    {
-                        _consoleBufferLines[currentLogicalLine].Append(c);
-                    }
+                    _consoleBufferLines[currentLogicalLine].Append(Options.ContinuationPrompt);
 
                     if (inSelectedRegion)
                     {
                         // Turn off inverse before end of line, turn on after continuation prompt
                         _consoleBufferLines[currentLogicalLine].Append(Options.SelectionColor);
                     }
                 }
+                else if (char.IsControl(charToRender))
+                {
+                    MaybeEmphasize(i, color);
+                    _consoleBufferLines[currentLogicalLine].Append('^');
+                    _consoleBufferLines[currentLogicalLine].Append((char)('@' + charToRender));
+                }
                 else
                 {
-                    if (char.IsControl(charToRender))
+                    MaybeEmphasize(i, color);
+                    _consoleBufferLines[currentLogicalLine].Append(charToRender);
+                }
+            }
+
+            if (inSelectedRegion)
+            {
+                _consoleBufferLines[currentLogicalLine].Append("\x1b[0m");
+            }
+
+            inSelectedRegion = false;
+            if (suggestion != null)
```

Comment:
This isn't necessary as it's not read anymore, but if you really want it, at least move it inside the previous `if` block that writes out the reset CSI.

Reply:
Oh, I see, it's needed because of the local function call to UpdateColorsIfNecessary - so just move it.

---
In reply to: [420571809](https://github.com/PowerShell/PSReadLine/pull/1496#discussion_r420571809) [](ancestors = 420571809)

Reply:
Moved to the if block.